features/netsocket/emac-drivers/TARGET_Cypress/COMPONENT_WHD/interface/WhdSTAInterface.cpp

```diff
@@ -355,6 +355,15 @@ nsapi_error_t WhdSTAInterface::disconnect()
     }
     whd_emac_wifi_link_state_changed(_whd_emac.ifp, WHD_FALSE);
 
+    // remove the interface added in connect
+    if (_interface) {
```

Comment:
it seems like this would prevent ever connecting again.

Reply:
@morser499 , local private member **_interface** is updated with the interface details in add_ehternet_interface()API call.
Hence  **_interface** is updated to NULL inside remove_ethernet_interface()API call.
As part of connect interface is updated and as part of disconnect interface is removed. Hence reconnecting is not blocked.

Reply:
Have verified multiple time's connect -> disconnect -> connect to ensure re-connection is successful.